### ISSUE TITLE
can't use qiling inside binrayninja (_PythonScriptingInstanceInput)

### ISSUE THREAD

**Issue by nitrogar** [Vector35/binaryninja-api#2653]
**Binary Ninja Version**
latest (2.4.3001)

**Describe the bug**
I was trying to make an emulation plugin using qiling framework but due to the stdin wrapper qiling dose not work
`  File "C:\Users\bhnx1\AppData\Local\Programs\Python\Python39\lib\site-packages\qiling\os\os.py", line 43, in __init__
    self.stdin  = ql_file('stdin',  sys.stdin.fileno())
AttributeError: '_PythonScriptingInstanceInput' object has no attribute 'fileno'`
**To Reproduce**
just try using fileno() on _PythonScriptingInstance

**Expected behavior**
return the file descriptor 

**Version and Platform (required):**
 - Binary Ninja: Dev 2.4.3001
 - OS: [Windows 10]

**Additional context**
the emulated program should be able to read from the wrapped stdin and write to the wrapped stdout

**Comment by psifertex** [comment]
At its root, this is basically a duplicate of #30 which is honestly something we might not ever do. Making a fully featured terminal would be a huge amount of work for the value it would provide versus just using an existing terminal.

In fact, if you look at the [code in question](https://github.com/qilingframework/qiling/blob/master/qiling/os/os.py#L48-L58) in qiling, this is directly where they are detecting whether they're running in the IDA Scripting console to adapt to work in that (also non-terminal) environment!

In short, for Qiling to work, it needs to be updated to work in our psuedo-terminal and I don't know enough about their architecture to know what that would entail but it's unfortunately not something we plan to fix.
